search-service/src/test/kotlin/com/egm/stellio/search/service/TemporalEntityAttributeServiceTests.kt

```diff
@@ -282,6 +283,164 @@ class TemporalEntityAttributeServiceTests : WithTimescaleContainer, WithKafkaCon
+                it.createdAt.isBefore(mergedAt) &&
+                it.modifiedAt == mergedAt
+        }
+    }
+
+    @Test
+    fun `it should merge an entity attributes`() = runTest {
+        val rawEntity = loadSampleData()
+
+        coEvery { attributeInstanceService.create(any()) } returns Unit.right()
+
+        temporalEntityAttributeService.createEntityTemporalReferences(rawEntity, listOf(APIC_COMPOUND_CONTEXT))
+            .shouldSucceed()
+
+        val mergedAt = ngsiLdDateTime()
+        val newProperties = loadSampleData("fragments/beehive_mergeAttributes.json")
+        val expandedAttributes = JsonLdUtils.expandAttributes(newProperties, listOf(APIC_COMPOUND_CONTEXT))
+        val ngsiLdAttributes = expandedAttributes.toMap().toNgsiLdAttributes().shouldSucceedAndResult()
+        temporalEntityAttributeService.mergeEntityAttributes(
+            beehiveTestCId,
+            ngsiLdAttributes,
+            expandedAttributes,
+            mergedAt,
+            null,
+            null
+        ).shouldSucceedWith {
+            it.updated.size == 6 &&
+                it.updated.filter { it.updateOperationResult == UpdateOperationResult.APPENDED }.size == 2 &&
+                it.updated.filter { it.updateOperationResult == UpdateOperationResult.APPENDED }
+                    .map { it.attributeName }
+                    .containsAll(listOf(OUTGOING_PROPERTY, TEMPERATURE_PROPERTY))
+        }
+
+        val newTeas = temporalEntityAttributeService.getForEntity(
+            beehiveTestCId,
+            emptySet()
+        )
+        assertEquals(6, newTeas.size)
+        coVerify {
+            attributeInstanceService.create(
+                match {
+                    it.value == null &&
+                        it.measuredValue == 1575.0 &&
+                        it.timeProperty == AttributeInstance.TemporalProperty.OBSERVED_AT &&
+                        it.time.isEqual(ZonedDateTime.parse("2022-01-24T14:01:22.066Z"))
+                }
+            )
+        }
+        // 1 when we create entity attributes and 3 when we merge entity attributes
+        coVerify(exactly = 4) {
+            attributeInstanceService.create(
+                match { it.timeProperty == AttributeInstance.TemporalProperty.OBSERVED_AT }
+            )
+        }
+        // 0 when we create entity attributes and 4 when we merge entity attributes
+        coVerify(exactly = 4) {
+            attributeInstanceService.create(
+                match { it.timeProperty == AttributeInstance.TemporalProperty.MODIFIED_AT }
+            )
+        }
+        // 4 when we create entity attributes and 2 when we merge entity attributes
+        coVerify(exactly = 6) {
+            attributeInstanceService.create(
+                match { it.timeProperty == AttributeInstance.TemporalProperty.CREATED_AT }
+            )
+        }
+    }
+
+    @Test
+    fun `it should merge an entity attributes with observedAt parameter`() = runTest {
```

Comment:
if you just want to test the behavior when an `observedAt` parameter is provided, better to use more focused test fixture. there are a lot of attributes in `beehive_mergeAttributes_without_observedAt.json` but it seems only the `incoming` property is tested, which makes the whole thing hard to understand.

maybe use fixture data with one property with an `observedAt` and another one without and test the two resulting TEAs?